internal/app/donor_store.go

```diff
@@ -198,6 +198,36 @@ func (s *donorStore) Put(ctx context.Context, lpa *page.Lpa) error {
 	return s.dynamoClient.Put(ctx, lpa)
 }
 
+func (s *donorStore) Delete(ctx context.Context) error {
+	data, err := page.SessionDataFromContext(ctx)
+	if err != nil {
+		return err
+	}
+
+	if data.SessionID == "" || data.LpaID == "" {
+		return errors.New("donorStore.Create requires SessionID and LpaID")
+	}
+
+	keys, err := s.dynamoClient.AllKeysByPk(ctx, lpaKey(data.LpaID))
+	if err != nil {
+		return err
+	}
+
+	canDelete := false
+	for _, key := range keys {
+		if key.PK == lpaKey(data.LpaID) && key.SK == donorKey(data.SessionID) {
```

Comment:
this is a bit defensive, but with it being a hard delete I think sensible

Reply:
Fair 👍